man/sd_bus_send.xml

```diff
@@ -68,6 +68,11 @@
     single receiver). It's behavior is similar to calling
     <citerefentry><refentrytitle>sd_bus_message_set_destination</refentrytitle><manvolnum>3</manvolnum></citerefentry>
     followed by calling <function>sd_bus_send()</function>.</para>
+
+    <para>Note that <function>sd_bus_send()</function>/<function>sd_bus_send_to()</function> might queue the
+    specified message locally only, in case the transport buffers are momentarily congested. Make sure to
+    invoke <citerefentry><refentrytitle>sd_bus_process</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    in time to flush the local queue into the transport buffers.</para>
```

Comment:
I don't understand this. Looking at `sd_bus_send`, it either calls `bus_write_message` which ends up calling `sendmsg`, or if the bus is not running, it pushes the messages onto `bus->wqueue` up to a certain limit. "queue locally ... when transport buffers are momentarily congested" doesn't match this. Also "in time to flush" is not clear: is this referring to the message timeout? (I would describe the behaviour as: If the bus connection is running, `sd_bus_send` passes the message to the kernel to be sent. If the connection is not established yet, it queues the message internally. If the connection is already closed or in in the process of being closed, the message is refused. `sd_bus_process` should be called...)

Reply:
Look more closely: bus_write_message() will try to write the msg to the socket buffer, but if they are full they might do a partial write (including no write at all) only, and will return how much it wrote in `&idx` parameter. If that happens we'll add the message to the local queue too. Basically, we always try to hand everything down to the sockets as much as we can, but if that doesn#t work, either because we are not set up properly yet, or the socket buffers are full we add what we need to do the queue instead, so that it is written out later.